### DISCUSSION TITLE
For async: show a loading indicator / state

### DISCUSSION THREAD

**Discussion by dennispaagman** [josefarias/hotwire_combobox#219] [Ideas]
To make clear to the user that data is loading.

I like how it's implemented in [React Select](https://react-select.com/async#loading-asynchronously): it shows a "loading" message in the combobox and has a loading indicator on the right side inside the input.

**Comment by josefarias** [comment]
I've gone back and forth on this. I think spinners can be a crutch in this case. My (possibly incorrect) take is most implementers should be able to return search results fast enough that the spinner isn't necessary. I do expect caching and SQL optimizations to be necessary in some cases.

I think if async loading is slow enough that it requires showing a spinner — that points to the implementer having to do some work to optimize fetching their results. And providing a spinner by default removes that positive pressure.

Open to having my mind changed though!

**Comment by matias-sclavi** [comment]
First of all, thank you for the amazing work on this!

I think it's similar to https://github.com/josefarias/hotwire_combobox/discussions/210 in the sense that could help with the UX out of the box. The scenario were you have pagination could also make use of this option: if you have choppy internet connection, it could feel like the list has already finished loading when in fact you still have items to be loaded.
